pandas/core/internals/managers.py

```diff
@@ -596,18 +596,22 @@ def replace_list(
         # figure out our mask apriori to avoid repeated replacements
         values = self.as_array()
 
-        def comp(s, regex=False):
+        def comp(s, mask, regex: bool = False):
```

Comment:
can you add the same annotation you have below for this? ArrayLike (can do same on s i think)

Reply:
@jreback done